fabric-recipe-api-v1/src/main/java/net/fabricmc/fabric/mixin/recipe/ingredient/IngredientMixin.java

```diff
@@ -35,9 +37,18 @@
 public class IngredientMixin implements FabricIngredient {
 	@Inject(method = "createCodec", at = @At("RETURN"), cancellable = true)
 	private static void injectCodec(boolean allowEmpty, CallbackInfoReturnable<Codec<Ingredient>> cir) {
-		final Codec<CustomIngredient> customIngredientCodec = allowEmpty ? CustomIngredientImpl.ALLOW_EMPTY_INGREDIENT_CODECS : CustomIngredientImpl.DISALLOW_EMPTY_INGREDIENT_CODECS;
-		Codec<Ingredient> ingredientCodec = customIngredientCodec.xmap(CustomIngredient::toVanilla, FabricIngredient::getCustomIngredient);
-		cir.setReturnValue(CustomIngredientImpl.first(cir.getReturnValue(), ingredientCodec));
```

Comment:
1. This is the only use of `CustomIngredientImpl#first`. Consider removing it if it's no longer used?
2. I think the fix is as simple as flipping the two here.

Reply:
No, we don't want to risk skipping the `type` field when reading IMO (e.g. there could be a custom ingredient with an `item` field). I meant to remove the codec but forgot, now done.